### ISSUE TITLE
Deleting a transaction brings you back out to the Dashboard rather than remaining in the account view.

### ISSUE THREAD

**Issue by n-stein** [moneymanagerex/moneymanagerex#6456]
... if you hit the delete button at the bottom to delete a transaction rather than crashing. It brings you back out to the Dashboard after deleting the transaction.

I'm not sure if that is intended behavior or not as intuitively I would think it should just delete the transaction and leave you in the account you are working in.

_Originally posted by @Ocirasai in https://github.com/moneymanagerex/moneymanagerex/issues/6139#issuecomment-1878920070_

**Comment by luisdev** [comment]
I agree that deleting a transaction should take you back to the account you were working in, not the Dashboard.

**Comment by n-stein** [comment]
This only seems to be experienced by a subset of users, I haven't been able to reproduce it on my Windows 10, Windows 11, Debian, or macOS dev systems.

**Comment by n-stein** [comment]
The exact same method (TransactionListCtrl::OnDeleteTransaction) is called for all three of these.

Maybe something with what control has focus? When you select a transaction and press the delete key the ListCtrl has focus, but if you press the delete button on the panel or use the right click menu those controls take focus from the ListCtrl.

**Comment by Ocirasai** [comment]
Is there a way in the method to make sure that focus is handed back to listCtrl before any other steps are executed?

**Comment by n-stein** [comment]
> Is there a way in the method to make sure that focus is handed back to listCtrl before any other steps are executed?

It should already be handled via this line:

https://github.com/moneymanagerex/moneymanagerex/blob/d7df17ead6d3fc81a898da4df6a90437cb8af223/src/mmchecking_list.cpp#L1646

**Comment by Ocirasai** [comment]
To try and alleviate any possible Microsoft Windows issue conflicting with Money Manager Ex, I did the following:

1. I performed a complete reset of Windows (including erasing the Windows drive). This returned me to a factory default Windows setting.
2. Next, I installed all of the Windows updates.
3. I then downloaded and installed a clean version 1.80 64-Bit for Windows Money Manager Ex release from the site.
4. Instead of using my old database, I created a new database and added a single account.
5. Next, I added one transaction to the account and then went back out to the Dashboard.
6. I went back into the account and deleted the transition - same behavior as before where I am dropped back to the Dashboard after the delete action.

I'm unsure where else to look or what to try. Starting with a clean Windows 10 System with just the Windows updates on it (I did not install any other software) should be sufficient for ensuring that I do not have anything that conflicts with Money Manager Ex.

Thank you.

- Oci
